app/services/CreditHistoryService.scala

```diff
@@ -45,16 +45,16 @@ class CreditHistoryService @Inject()(incomeTaxViewChangeConnector: IncomeTaxView
       case financialDetailsModel: FinancialDetailsModel =>
         val fdRes = financialDetailsModel.getPairedDocumentDetails().flatMap {
           case (document: DocumentDetail, financialDetail: FinancialDetail) =>
-            (financialDetail.isMFACredit, document.credit.isDefined, financialDetail.isBalancingChargeCredit) match {
-              case (true, true, false) =>
+            (financialDetail.getCreditType, document.credit.isDefined) match {
+              case (Some(MfaCreditType), true) =>
                 Some(CreditDetailModel(date = document.documentDate, document, MfaCreditType, Some(financialDetailsModel.balanceDetails)))
-              case (false, true, true) =>
+              case (Some(BalancingChargeCreditType), true) =>
                 Some(CreditDetailModel(date = document.documentDate, document, BalancingChargeCreditType, Some(financialDetailsModel.balanceDetails)))
-              case (false, true, false) =>
+              case (_, true) =>
```

Comment:
I think we can put in this case:
`case ( Some(CutOverCreditType), true) => ...`